jobs/jobs_utils.py

```diff
@@ -31,6 +34,9 @@ def clone_model(model, **new_values):
     original entity, except where overridden. By default, it will have no
     parent entity or key name, unless supplied.
 
+    IMPORTANT: This function should be used in EVERY DoFn beacse beam expects
```

Comment:
Does/Can this happen "by default"? What if developers do not see this message?

Can we have a backend test that verifies it? (e.g. runs each job and verifies this function is called)

Reply:
I think that metaclass would be nice.

Reply:
I've spent a couple of hours writing one, but it's pretty involved and will require a lot of refactoring, so I don't want to include it in this PR. Let's track it in #12449 instead.